euphoria-flink/src/main/java/cz/seznam/euphoria/flink/batch/ReduceByKeyTranslator.java

```diff
@@ -159,16 +172,64 @@ public DataSet translate(FlinkOperator<ReduceByKey> operator,
     }
 
     @Override
-    public BatchElement<Window, Pair>
-    reduce(BatchElement<Window, Pair> p1, BatchElement<Window, Pair> p2) {
-
-      Window wid = p1.getWindow();
-      return new BatchElement<>(
-              wid,
-              Math.max(p1.getTimestamp(), p2.getTimestamp()),
-              Pair.of(
-                      p1.getElement().getFirst(),
-                      reducer.apply(Arrays.asList(p1.getElement().getSecond(), p2.getElement().getSecond()))));
+    public void combine(Iterable<BatchElement<Window, Pair>> values, Collector<BatchElement<Window, Pair>> out) {
+      doReduce(values, out);
+    }
+
+    @Override
+    public void reduce(Iterable<BatchElement<Window, Pair>> values, Collector<BatchElement<Window, Pair>> out) {
+      doReduce(values, out);
+    }
+
+    private void doReduce(Iterable<BatchElement<Window, Pair>> values,
+                       org.apache.flink.util.Collector<BatchElement<Window, Pair>> out) {
+
+      // Tuple2[Key, Window] => Reduced Value
+      Map<Tuple2, TimestampedElement> reducedValues = new HashMap<>();
+
+      for (BatchElement<Window, Pair> batchElement : values) {
+        Object key = batchElement.getElement().getFirst();
+        Window window = batchElement.getWindow();
+
+        Tuple2 kw = new Tuple2<>(key, window);
```

Comment:
may i ask for a short comment here explaining why the order is reversed compared to the order of the two items in `RBKKeySelector`; i know its in purpose; but i fear i'll forget about it in half a year.